### DISCUSSION TITLE
Program not saved in my Spike Hub

### DISCUSSION THREAD

**Discussion by nrtech1** [pybricks/support#1641] [Q&A]
Hi everyone
I am connecting to hub with PyBricks using bluetooth. I run the program no problem. When I disconnect the bluetooth the hub has no program to run. Am I doing something wrong?
Thank you

**Comment by dlech** [comment]
Maybe the program crashes when not connected with Bluetooth?

Can you share the program?

**Reply by BertLindeman** [reply]
The program is saved on hub-power-off.
You could try **before** you disconnect bluetooth to long press the button on the hub to do the power-off.

I just did this with success on a Technic hub.

Then power-on the hub by a button press and press again to start the last saved program.

Bert

**Comment by nrtech1** [comment]
Thank you bothunless I am mistaken is there any way other than bluetooth to establish a connection? I never tried wired. I am not at home so dont have access to my code but its a very small program to just make the motors move forward. Nothing more than that.
I will try what Bert commented and give my feedback later on todayhave a great dayNick
    On Tuesday, May 21, 2024 at 11:58:58 p.m. EDT, David Lechner ***@***.***> wrote:  
 
 


Maybe the program crashes when not connected with Bluetooth?

Can you share the program?
